### PR TITLE
Rapid-PCR test support (EXPOSUREAPP-11698, 11697, 11699)

### PR DESCRIPTION
**Pull request by BMItr**
This PR addresses Exposureapp-11252.

Feature: Support for Rapid-PCR tests.

DRAFT for orientation.

How to test:
- connect against the mock server
- generate a rapid-pcr QR  (_e.g.  cwa rpcr -e wru_) or similar

todos / inProgress:

- [x] add all relevant testcases
- [x] UI/Flow adjustments

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/rapid/RapidQrCodeExtractor.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    protected abstract val loggingTag: String
+    protected abstract fun String.removeQrCodePrefix(): String
+    protected abstract fun CleanPayload.toCoronaTestQRCode(rawString: String): CoronaTestQRCode.Rapid
+
+    override suspend fun extract(rawString: String): CoronaTestQRCode.Rapid {
+        Timber.tag(loggingTag).v("extract(rawString=%s)", rawString)
+        val payload = CleanPayload(extractData(rawString))
+
+        RatQrCodeCensor.dataToCensor = RatQrCodeCensor.CensorData(
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/rapid/RapidQrCodeExtractor.kt]: true.
naming adjusted

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/rapid/RapidQrCodeExtractor.kt [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import de.rki.coronawarnapp.qrcode.scanner.QrCodeExtractor
+import de.rki.coronawarnapp.util.HashExtensions.toSHA256
+import de.rki.coronawarnapp.util.hashing.isSha256Hash
+import de.rki.coronawarnapp.util.serialization.fromJson
+import okio.internal.commonToUtf8String
+import org.joda.time.Instant
+import org.joda.time.LocalDate
+import timber.log.Timber
+
+abstract class RapidQrCodeExtractor : QrCodeExtractor<CoronaTestQRCode> {
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/rapid/RapidQrCodeExtractor.kt:18]: I simply kept the naming scheme from before. Feel free to adjust it, if you want.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/rapid/RapidAntigenQrCodeExtractor.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        dateOfBirth = dateOfBirth,
+        testId = testId,
+        salt = salt,
+        isDccSupportedByPoc = isDccSupportedByPoc,
+        rawQrCode = rawString
+    )
+
+    companion object {
+        private const val PREFIX1: String = "https://s.coronawarn.app?v=1#"
+        private const val PREFIX2: String = "https://s.coronawarn.app/?v=1#"
```
[the remark this answers is not in the document]
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/rapid/RapidAntigenQrCodeExtractor.kt:31]: added

**Comment by mtwalli** [comment]
Two issues I found: 
1- Deeplinking does not register the test, it just opens the App, 
2- Positive RPCR test trigger the submission flow, even if it was submitted before (Happens from Recycle Bin)
  - Register a positive RPCR test and proceed with submission flow 
  - Register a PCR, TAN and proceed with duplicate confirmation 
  - Go to recycle bin and restore the recycled RPCR Test 
  - Notice that the submission flow is started again (it should not, since the test was already submitted)

**Comment by BMItr** [comment]
> Two issues I found: 1- Deeplinking does not register the test, it just opens the App, 2- Positive RPCR test trigger the submission flow, even if it was submitted before (Happens from Recycle Bin)
> 
> * Register a positive RPCR test and proceed with submission flow
> * Register a TAN for PCR test and proceed with duplicate confirmation
> * Go to recycle bin and restore the recycled RPCR Test
> * Notice that the submission flow is started again (it should not, since the test was already submitted)

@mtwalli  DeepLinking support added. Point 2 seems to me a general issue. (->Teams)

**Review by mtwalli — DISMISSED** [review]

**Review by mtwalli — APPROVED** [review]

**Review by jurajkusnier — APPROVED** [review]
LGTM

**Pull request merged**